src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilderHelper.cs

```diff
@@ -70,6 +70,7 @@ public async IAsyncEnumerable<ProjectAnalysisResult> BuildProjectIncremental()
                 {
                     string solutionFilePath = NormalizePath(WorkspacePath);
                     SolutionFile solutionFile = SolutionFile.Parse(solutionFilePath);
+                    var reverse = solutionFile.ProjectsInOrder.Reverse().ToList<ProjectInSolution>().AsReadOnly();
```

Comment:
@cslong  I tried multiple ways to reverse "`solutionFile.ProjectsInOrder`" , however the following line 163 will be failed on the second project during the foreach loop. 
`IAnalyzerResult analyzerResult = projectAnalyzer.Build(GetEnvironmentOptions(requiresNetFramework, projectAnalyzer.ProjectFile.ToolsVersion)).FirstOrDefault();`

Not sure if it's my end issue or if you can reproduce the problem.   I got the exception "ObjectDisposedException: Cannot write to a closed TextWriter."

System.AggregateException
  HResult=0x80131500
  Message=An error occurred while writing to logger(s). (Cannot write to a closed TextWriter.)
  Source=Microsoft.Extensions.Logging
  StackTrace:
   at Microsoft.Extensions.Logging.Logger.ThrowLoggingError(List`1 exceptions)
   at Microsoft.Extensions.Logging.Logger.Log[TState](LogLevel logLevel, EventId eventId, TState state, Exception exception, Func`3 formatter)
   at Microsoft.Extensions.Logging.Logger`1.Microsoft.Extensions.Logging.ILogger.Log[TState](LogLevel logLevel, EventId eventId, TState state, Exception exception, Func`3 formatter)
   at Microsoft.Extensions.Logging.LoggerExtensions.Log(ILogger logger, LogLevel logLevel, EventId eventId, Exception exception, String message, Object[] args)
   at Microsoft.Extensions.Logging.LoggerExtensions.Log(ILogger logger, LogLevel logLevel, String message, Object[] args)
   at Microsoft.Extensions.Logging.LoggerExtensions.LogDebug(ILogger logger, String message, Object[] args)
   at Buildalyzer.Environment.ProcessRunner.Start()
   at Buildalyzer.ProjectAnalyzer.BuildTargets(BuildEnvironment buildEnvironment, String targetFramework, String[] targetsToBuild, AnalyzerResults results)
   at Buildalyzer.ProjectAnalyzer.Build(String targetFramework, BuildEnvironment buildEnvironment)
   at Buildalyzer.ProjectAnalyzer.Build(String targetFramework, EnvironmentOptions environmentOptions)
   at Buildalyzer.ProjectAnalyzer.Build(EnvironmentOptions environmentOptions)
   at Codelyzer.Analysis.Build.WorkspaceBuilderHelper.BuildIncremental(String projectPath) in C:\git\codelyzer\src\Analysis\Codelyzer.Analysis.Build\WorkspaceBuilderHelper.cs:line 163
   at Codelyzer.Analysis.Build.WorkspaceBuilderHelper.RunTask(String projectPath) in C:\git\codelyzer\src\Analysis\Codelyzer.Analysis.Build\WorkspaceBuilderHelper.cs:line 129
   at Codelyzer.Analysis.Build.WorkspaceBuilderHelper.<>c__DisplayClass22_1.<BuildProjectIncremental>b__0() in C:\git\codelyzer\src\Analysis\Codelyzer.Analysis.Build\WorkspaceBuilderHelper.cs:line 85
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__272_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)